provider/rotation_resource.go

```diff
@@ -0,0 +1,483 @@
+									AttributePath: path,
+								},
+							}
+						}
+
+						return nil
+					},
+				),
+				DiffSuppressFunc: func(k, old, new string, d *schema.ResourceData) bool {
+					return true
+				},
+			},
+		},
+	}
+}
+
+func readResourceFireHydrantRotation(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	firehydrantAPIClient := m.(firehydrant.Client)
+
+	id := d.Id()
+	teamID := d.Get("team_id").(string)
+	scheduleID := d.Get("schedule_id").(string)
+	tflog.Debug(ctx, "Read rotation", map[string]interface{}{
+		"id":          id,
+		"team_id":     teamID,
+		"schedule_id": scheduleID,
+	})
+
+	rotation, err := firehydrantAPIClient.Rotations().Get(ctx, teamID, scheduleID, id)
+	if err != nil {
+		if errors.Is(err, firehydrant.ErrorNotFound) {
+			tflog.Debug(ctx, "Rotation %s does not exist", map[string]interface{}{
+				"id":          id,
+				"team_id":     teamID,
+				"schedule_id": scheduleID,
+			})
+			d.SetId("")
+			return nil
+		}
+		return diag.Errorf("Error reading rotation %s: %v", id, err)
+	}
+
+	outputMemberIDs := []string{}
+	memberIDs := rotation.Members
+	for _, memberID := range memberIDs {
+		if v := memberID.ID; v != "" {
+			outputMemberIDs = append(outputMemberIDs, memberID.ID)
+		}
+	}
+
+	attributes := map[string]interface{}{
+		"name":                               rotation.Name,
+		"time_zone":                          rotation.TimeZone,
+		"description":                        rotation.Description,
+		"enable_slack_channel_notifications": rotation.EnableSlackChannelNotifications,
+		"prevent_shift_deletion":             rotation.PreventShiftDeletion,
+		"color":                              rotation.Color,
+		"members":                            outputMemberIDs,
+		"strategy":                           rotationStrategyToMap(rotation.Strategy),
+		"restrictions":                       rotationRestrictionsToData(rotation.Restrictions),
+	}
+	if rotation.SlackUserGroupID != "" {
+		attributes["slack_user_group_id"] = rotation.SlackUserGroupID
+	}
+	if rotation.CoverageGapNotificationInterval != "" {
+		attributes["coverage_gap_notification_interval"] = rotation.CoverageGapNotificationInterval
+	}
+	if rotation.StartTime != "" {
+		attributes["start_time"] = rotation.StartTime
+	}
+
+	for key, value := range attributes {
+		if err := d.Set(key, value); err != nil {
+			return diag.Errorf("Error setting %s for rotation %s: %v", key, id, err)
+		}
+	}
+
+	d.SetId(rotation.ID)
+
+	return diag.Diagnostics{}
+}
+
+func createResourceFireHydrantRotation(ctx context.Context, d *schema.ResourceData, m interface{}) diag.Diagnostics {
+	firehydrantAPIClient := m.(firehydrant.Client)
+
+	teamID := d.Get("team_id").(string)
+	scheduleID := d.Get("schedule_id").(string)
+	tflog.Debug(ctx, fmt.Sprintf("Create rotation: %s", teamID), map[string]interface{}{
+		"team_id":     teamID,
+		"schedule_id": scheduleID,
+	})
+
+	inputMemberIDs := d.Get("members").([]interface{})
```

Comment:
I think `members` can be `nil` so this would panic. Should we have more checks or am I misunderstanding?
```suggestion
	inputMemberIDs, ok := d.Get("members").([]interface{})
	if !ok {
	  ... ¯\_(ツ)_/¯
	}
```

Reply:
TIL!